libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs

```diff
@@ -120,17 +122,25 @@ public override async Task<DialogTurnResult> ResumeDialogAsync(DialogContext dc,
             // To avoid the prompt prematurely ending we need to implement this method and
             // simply re-prompt the user.
             await RepromptDialogAsync(dc.Context, dc.ActiveDialog, cancellationToken).ConfigureAwait(false);
-            return Dialog.EndOfTurn;
+            return EndOfTurn;
         }
 
         public override async Task RepromptDialogAsync(ITurnContext turnContext, DialogInstance instance, CancellationToken cancellationToken = default(CancellationToken))
         {
             var state = (IDictionary<string, object>)instance.State[PersistedState];
             var options = (PromptOptions)instance.State[PersistedOptions];
-            await OnPromptAsync(turnContext, state, options, cancellationToken).ConfigureAwait(false);
+            await OnPromptAsync(turnContext, state, options, false, cancellationToken).ConfigureAwait(false);
```

Comment:
Based on what I see in js, I think retry should be true for RepromptDialogAsync:
https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-dialogs/src/prompts/activityPrompt.ts#L98